irispy/iris_tools.py

```diff
@@ -24,14 +24,20 @@
 # paper.
 DETECTOR_GAIN = {"NUV": 18., "FUV": 6., "SJI": 18.}
 DETECTOR_YIELD = {"NUV": 1., "FUV": 1.5, "SJI": 1.}
-BSCALE = 0.25
-BZERO = 7992
+SJI_DEFAULT_BSCALE = 0.25
+SJI_DEFAULT_BZERO = 7992.0
 DN_UNIT = {
     "NUV": u.def_unit("DN_IRIS_NUV", DETECTOR_GAIN["NUV"]/DETECTOR_YIELD["NUV"]*u.photon),
     "FUV": u.def_unit("DN_IRIS_FUV", DETECTOR_GAIN["FUV"]/DETECTOR_YIELD["FUV"]*u.photon),
     "SJI": u.def_unit("DN_IRIS_SJI", DETECTOR_GAIN["SJI"]/DETECTOR_YIELD["SJI"]*u.photon),
     "SJI_UNSCALED": u.def_unit("DN_IRIS_SJI_UNSCALED",
-                               (DETECTOR_GAIN["SJI"]/DETECTOR_YIELD["SJI"]*BSCALE + BZERO)*u.photon)}
+                               (DETECTOR_GAIN["SJI"]/DETECTOR_YIELD["SJI"]*SJI_DEFAULT_BSCALE
+                                + SJI_DEFAULT_BZERO)*u.photon)}
+SJI_SCALING = ((DN_UNIT["SJI"],
+                DN_UNIT["SJI_UNSCALED"],
+                lambda x: (x - SJI_DEFAULT_BZERO) / SJI_DEFAULT_BSCALE,
+                lambda x: x * SJI_DEFAULT_BSCALE + SJI_DEFAULT_BZERO))
```

Comment:
I modified the unit to what you suggest but it is still not working ...

On the terminal, I wrote : 
`iris_tools.DN_UNIT['SJI'].to(iris_tools.DN_UNIT['SJI_UNSCALED'], equivalencies=iris_tools.SJI_SCALING)`

Is it the good way to use `equivalencies` ?

My return is always `- 0.000564334085778781`

Reply:
Hi @BaptistePellorceAstro.  I've figured out what the problem is.  In fact there are a two:
* Firstly, your definition of the ```DN_IRIS_SJI_UNSCALED``` unit in L34-35 is still incorrectly inverted.
* Secondly, ```DN_IRIS_SJI``` and ```DN_IRIS_SJI_UNSCALED``` are currently both defined as a fixed number of photons.  This means they are already compatible units and so the equivalency is not being applied as we want.  The fix to this is set ```DN_IRIS_SJI_UNSCALED``` to counts (```u.ct```) instead of photons (```u.photon```) in L35.
This should fix it.  Let me know if this works for you.

Reply:
Hi, for the expression, I have changed it in my file that is not update on github. Then, I test with `u.ct` and I have `'DN_IRIS_SJI' and 'DN_IRIS_SJI_UNSCALED' are not convertible`. 

Tiago has found a way to avoid the problem : creating the unit as  
`DN_UNIT = {"SJI": u.def_unit("DN_IRIS_SJI", DETECTOR_GAIN/DETECTOR_YIELD*u.photon),
                     "SJI_UNSCALED": u.def_unit("DN_IRIS_SJI_UNSCALED")}`

Then the unit is just a name but the equivalencies are working.

Maybe, I must set the `DN_UNIT["SJI"]` for everyone and convert data with memmap with the equivalencies function.

Reply:
This pull request is getting really long and hard to follow. I have never used the `units` package, and so far fail to see the usefulness of writing an equivalency that will require a user to type an extremely long line of code. These units are meaningless in any case. I'd just set the units to `SJI_UNSCALED` or `SJI_SCALED` and be done with it. If people want `SJI_SCALED`, don't use `memmap`. This is what IDL does and I don't see why we should spend so long thinking about a feature that very few will ever use.

Reply:
I agree this PR has gotten far too big.  We should merge it as soon as possible.

Regarding the ``DN_UNIT```, I like @tiagopereira's solution.  Let's go with that and then get this PR merged.

I think we should keep the equivalency now that it has been written.  I foresee useful applications in the future.  Regarding units in general, they are a requirement of SunPy and AstroPy and so I think we stick to that convention to continue to leverage work from those packages.  As a user in my own work, I find units invaluable and I'm sure other users will find the same.  From what @BaptistePellorceAstro says, it sounds like he has made all the necessary changes regarding unit and equivalencies and just needs to push his latest changes before we can merge.

Reply:
Hi, so I will delete the expression of the unscaled unit as Tiago said and let the unit just with the name. I let it without any value for now ?

Reply:
Yes.  Define the unscaled unit with just a name and leave the ```SJI_SCALING``` as is, so long as it works.  It should.  Push those changes and then we can get it merged.

Reply:
Ok, merging will be available at the european morning tomorrow.

Reply:
I can't set no value in the `SJI_UNSCALED` unit because `uncertainty` is using it. I so input `u.photon` as value to avoid an error.